### PR TITLE
Index page add tiles

### PR DESCRIPTION
**Pull request by dhirajshettymoj**
Code updates to all tiles to the index page based on roles.

### CODE REVIEW

File: server/middleware/populateCurrentUserRoles.ts [hunk 1]
```diff
@@ -0,0 +1,12 @@
+import { RequestHandler } from 'express'
+import jwtDecode from 'jwt-decode'
+
+export default function populateCurrentUserRoles(): RequestHandler {
+  return async (req, res, next) => {
+    if (res.locals.user?.token) {
+      const { authorities: roles } = jwtDecode(res.locals.user.token) as { authorities?: string[] }
+      res.locals.user = { roles, ...res.locals.user }
```
Comment (tpmcgowan) [inline server/middleware/populateCurrentUserRoles.ts:8]: I guess you could just include this line in `server/middleware/authorisationMiddleware.ts` where the token is already decoded and then you wouldn't need to have this file at all?

But, appreciate that it it makes it clearer what is going on, alongside the `populateCurrentUser` middleware...

File: assets/js/page-enhancements.js [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+  const extractCsrfFieldAsKeyValuePairFromForm = form => {
+    return form
+      .serializeArray()
+      .filter(formField => formField.name === '_csrf')
+      .map(formField => {
+        return `${formField.name}=${formField.value}`
+      })
+  }
+
+  const submitCookiePreferencesFormViaAjax = () => {
```
Comment (tpmcgowan) [inline assets/js/page-enhancements.js:103]: You mightn't need all of these functions - e.g. is cookie preferences used in the staff app?

**Review by tpmcgowan — APPROVED** [review]
LGTM - some minor suggestions but happy to approve 👍

File: server/middleware/populateCurrentUserRoles.ts (hunk 1 above)
Comment (dhirajshettymoj) [reply server/middleware/populateCurrentUserRoles.ts:8]: will leave it in for now.

File: assets/js/page-enhancements.js (hunk 2 above)
Comment (dhirajshettymoj) [reply assets/js/page-enhancements.js]: Agreed. Removed the public facing methods which will not be used.

**Review by tpmcgowan — APPROVED** [review]

**Pull request merged**